src/fmu/sumo/explorer/objects/cpgrid_property.py

```diff
@@ -59,25 +59,35 @@ def grid(self):
         Returns:
             Grid: a Grid object (an instance of class CPGrid).
         """
-        sc = SearchContext(self._sumo).filter(
+        sc = SearchContext(self._sumo).grids.filter(
+            uuid=self.caseuuid,
+            iteration=self.iteration,
+            realization=self.realization,
+        )
+        should = [
+            {"term": {"data.name.keyword": self.tagname}},
+        ]
+        if (
+            self.metadata.get("data", {})
+            .get("geometry", {})
+            .get("relative_path", None)
+            is not None
+        ):
+            should.append(
+                {
+                    "term": {
+                        "file.relative_path.keyword": self._metadata["data"][
```

Comment:
What's the difference between `self.metadata` and `self._metadata`?

Reply:
None; `self.metadata` is an accessor for `self._metadata`... we (I) should be consistent about this... I'll update the PR.